app/models/anime.rb

```diff
@@ -126,82 +181,89 @@ def self.sfw_filter(current_user)
     end
   end
 
-  # Check whether the current anime is SFW.
-  def sfw?
-    age_rating != "R18+"
+  def poster_image_thumb
+    if poster_image_file_name.nil?
+      'https://hummingbird.me/assets/missing-anime-cover.jpg'
+    else
+      # This disgusting fastpath brought to you by the following issue:
+      # https://github.com/thoughtbot/paperclip/issues/909
+      if Rails.env.production?
+        url = 'https://static.hummingbird.me/anime/poster_images'
+        id_path = format('%03d/%03d/%03d',
+                         (id / 1_000_000 % 1_000),
+                         (id / 1_000 % 1_000),
+                         (id % 1_000))
+        file_name = URI.escape(poster_image_file_name.rpartition('.')[0])
+        updated = poster_image_updated_at.to_i
+        "#{url}/#{id_path}/large/#{file_name}.jpg?#{updated}"
+      else
+        poster_image.url(:large)
+      end
+    end
   end
 
-  def self.order_by_rating
-    order('bayesian_rating DESC NULLS LAST, user_count DESC NULLS LAST')
+  before_validation do
+    self.cover_image_top_offset = 0 if cover_image_top_offset.nil?
+  end
+
+  # Check whether the current anime is SFW.
+  def sfw?
+    age_rating != 'R18+'
   end
 
   # Use this function to get the title instead of directly accessing the title.
-  def canonical_title(title_language_preference=nil)
-    if title_language_preference and title_language_preference.class == User
-      title_language_preference = title_language_preference.title_language_preference
+  def canonical_title(preference = '')
+    if preference.class == User
+      preference = preference.title_language_preference
     end
 
-    if title_language_preference and title_language_preference == "romanized"
-      return title
-    elsif title_language_preference and title_language_preference == "english"
-      return (alt_title and alt_title.length > 0) ? alt_title : title
+    if (preference == 'english' || english_canonical) && !alt_title.nil?
+      alt_title
     else
-      return (alt_title and alt_title.length > 0 and english_canonical) ? alt_title : title
+      title
     end
   end
 
   # Use this function to get the alt title instead of directly accessing the
   # alt_title.
-  def alternate_title(title_language_preference=nil)
-    if title_language_preference and title_language_preference.class == User
-      title_language_preference = title_language_preference.title_language_preference
+  def alternate_title(preference = '')
+    if preference.class == User
+      preference = preference.title_language_preference
     end
 
-    if title_language_preference and title_language_preference == "romanized"
-      return alt_title
-    elsif title_language_preference and title_language_preference == "english"
-      return (alt_title and alt_title.length > 0) ? title : nil
+    if preference == 'english' || english_canonical
```

Comment:
I _think_ you can access boolean fields as if they were predicate methods so `english_canonical?` might be more conventional